src/main/java/com/fasterxml/jackson/core/json/UTF8JsonGenerator.java

```diff
@@ -647,11 +647,16 @@ public void writeUTF8String(byte[] text, int offset, int len) throws IOException
             _flushBuffer();
         }
         _outputBuffer[_outputTail++] = _quoteChar;
+
+        // When writing raw UTF-8 encoded bytes, it is beneficial if the escaping table can directly be indexed into
+        // using the byte value.
+        final int[] extendedOutputEscapes = _extendOutputEscapesTo8Bits();
```

Comment:
Ok... this is the (only) part I find problematic. Having to dynamically change `_outputEscapes` seems problematic, although I understand why it is being done.  Since it is something that may be changed by a call to `JsonGenerator.setCharacterEscapes()` modifications cannot be done on constructor.

But: I have an idea for bit bigger changes that would make it possible to eagerly ensure `_outputEscapes` is 256 elements long. Will add a separate comment.

Reply:
Actually, scratch that. Only now realized this is limited to `writeUTF8String`, not all escaping.

So dynamically copying + changing is actually reasonable since it's not always needed etc.